### PR TITLE
[READY] Bluespace entropy

### PR DESCRIPTION
**Pull request by Evankhell561**
## About The Pull Request
Add several new features. Among which are a new mob, Bluespace roaches, a garbage spawn event, and one to spawn a rare item due to players exceeding the bluespace entropy limit.

---
image

thanks to Deon and Infrared for their sprites.
## Changelog
:cl:
add: Bluespace entropy
add: bluespace distrosion events.
add: mob: stranger.
add: Bluespace roaches
/:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/energy/plasma.dm [hunk 1]
```diff
@@ -102,4 +102,21 @@
 	overlays.Cut()
 	..()
 	if(cell)
-		overlays += image(icon, "cell_guild")
\ No newline at end of file
+		overlays += image(icon, "cell_guild")
+
+/obj/item/weapon/gun/energy/plasma/stranger
```
Comment (SpadesSlick) [inline code/modules/projectiles/guns/energy/plasma.dm:107]: It should have it's own file, along with it's own sprite.

File: code/__HELPERS/bluespace_entropy.dm [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	else
+		GLOB.bluespace_entropy += rand(0, max_value)
+		var/entropy_cap = rand(100, 300)
+		if(GLOB.bluespace_entropy >= entropy_cap)
+			bluespace_distorsion(T, minor_dristortion)
+			GLOB.bluespace_entropy -= entropy_cap
+
+/proc/bluespace_distorsion(turf/T, minor_dristortion=FALSE)
+	var/distortion_value = rand(1, 100)
+	switch(distortion_value)
```
Comment (SpadesSlick) [inline code/__HELPERS/bluespace_entropy.dm:20]: Bluespace crystals were added recently, so you can add it here.

File: code/__HELPERS/bluespace_entropy.dm [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+/proc/bluespace_gift(turf/T, minor_dristortion)
+	var/amount = 1
+	var/chance_to_second = rand(2,10)
+	if(minor_dristortion)
+		chance_to_second = round(chance_to_second/2)
+	if(prob(chance_to_second))
+		amount = amount * 2
+	for(var/i=1 to amount)
+		var/obj/O = pickweight(RANDOM_RARE_ITEM - /obj/item/stash_spawner)
```
Comment (SpadesSlick) [inline code/__HELPERS/bluespace_entropy.dm:73]: It supposed to spawn special oddity with unique sprites, not this.

File: code/__HELPERS/bluespace_entropy.dm (hunk 2 above)
Comment (Evankhell561) [reply code/__HELPERS/bluespace_entropy.dm:20]: These already contribute to increasing entropy when used to teleport something.

Comment (SpadesSlick) [reply code/__HELPERS/bluespace_entropy.dm:20]: I mean, they are blue space related event.

**Comment by GrayRachnid** [comment]
Is it possible to get an explanation on what exactly is the bluespace entropy feature?

**Comment by Evankhell561** [comment]
> Is it possible to get an explanation on what exactly is the bluespace entropy feature?


every time someone teleports the bluespace entropy level increases. When this reaches a limit, a random event related to bluespace happens.

**Comment by SpadesSlick** [comment]
> Is it possible to get an explanation on what exactly is the bluespace entropy feature?

If in short - more small mid round events that are not tied to storyteller, to make rounds less stale.

**Review by Tea-Primoris — APPROVED** [review]
Everything seems to be working.

**Comment by Firefox13** [comment]
unkown............
изображение

**Review by TorinoFermic — APPROVED** [review]

**Comment by TorinoFermic** [comment]
There's tons of conflicts, please resolve before I can merge it.

**Review by Hopekz — APPROVED** [review]
Looks good here direction-wise.
I like how the bluespace roaches kind of "jump" when they fight you.

**Review by TorinoFermic — APPROVED** [review]

**Comment by Hopekz** [comment]
Woo

**Pull request merged**